File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Deque<Integer> queue = new ArrayDeque<>();
        for (int i = 0; i < progresses.length; i++) {
            int days = (int) Math.ceil((double) (100 - progresses[i]) / speeds[i]);
            queue.offer(days);
        }

        List<Integer> list = new ArrayList<>();
        while (!queue.isEmpty()) {
            int standard = queue.poll();
            int count = 1;

            while (!queue.isEmpty() && queue.peek() <= standard) {
                queue.poll();
                count++;
            }

            list.add(count);
        }

        int[] answer = new int[list.size()];
        for (int i = 0; i < answer.length; i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
import math
def solution(progresses, speeds):
    answer = []
    queue = deque()
    for p, s in zip(progresses, speeds):
        days = math.ceil((100-p) / s)
        queue.append(days)

    while queue:
        now = queue.popleft()
        count = 1
        while queue and now >= queue[0]:
            queue.popleft()
            count += 1
        answer.append(count)
    return answer